packages/x-charts/src/internals/plugins/utils/ChartStore.ts

```diff
@@ -37,8 +37,9 @@ export class ChartStore<
   public update = (updater: StoreUpdater<TSignatures, TOptionalSignatures>) => {
     const newState = updater(this.value);
     if (newState !== this.value) {
+      const prevValue = this.value;
       this.value = newState;
-      this.listeners.forEach((l) => l(newState));
```

Comment:
Do we want to test `useSelectorEffect` as implemented in [this PR](https://github.com/mui/mui-x/pull/18275/files#diff-8341a10e08f545bf13be5fdb23293fd7643b4b3bcc4116e0f2441e5e056d0b5a)?

Reply:
I gave it a try. The implementation of the effect is effectively simpler, and avoid having to keep in memory the entire store.

I just needed to modify the implementation to creat ea `useLazySelectorEffect`.

The issue is that if `onAxisInteraction` is not provided, we can do an early return in the effect. But we are still calling the selector.
The proposed solution add a `skip` parameter that allows to remove the selector computation if the effect is not needed